doc/configuration.md

```diff
@@ -0,0 +1,247 @@
+- **Environment variable**: `DD_TRACE_AGENT_URL`
+- **Default value**: `""`
+
+### Service Name
+The default service name to associate with spans produced by the tracer.
+Service name can be overridden programmatically on a per-span basis via the
+`datadog::opentracing::Span::setServiceName` member function.
+
+- **TracerOptions member**: `std::string service`
+- **JSON property**: `"service"` _(string)_
+- **Environment variable**: `DD_SERVICE`
+- **Required**
+
+### Service Type
+The default "service type" to associate with spans produced by the tracer.
+
+Service type is used in multiple places throughout Datadog to distinguish
+different categories of instrumented service from each other.  For example,
+it is used in the following ways:
+
+- to identify whether the service's spans need to be obfuscated
+- to control display of the service in the Datadog UI.
+
+Example values for service type are `web`, `db`, and `lambda`.
+
+- **TracerOptions member**: `std::string type`
+- **JSON property**: `"type"` _(string)_
+- **Default value**: `"web"`
+
+### Environment
+The default release environment in which the service is running, e.g. "staging"
+or "prod".
+
+- **TracerOptions member**: `std::string environment`
+- **JSON property**: `"environment"` _(string)_
+- **Environment variable**: `DD_ENV`
+- **Default value**: `""`
+
+### Sampling Rate
+The default probability that a trace beginning with this tracer will be sampled
+for ingestion.
+  
+For more information about the configuration of trace sampling, see
+[sampling.md][6].
+
+- **TracerOptions member**: `double sample_rate`
+- **JSON property**: `"sample_rate"` _(number)_
+- **Environment variable**: `DD_TRACE_SAMPLE_RATE`
+
+### Sampling Rules
+Sampling rules allow for fine-grained control over the rate at which traces
+beginning at this tracer will be sampled for ingestion.  Sampling rules are
+specified as a JSON array of objects.
+
+For more information about the configuration of trace sampling, see
+[sampling.md][6].
+
+- **TracerOptions member**: `std::string sampling_rules`
+- **JSON property**: `"sampling_rules"` _(array of objects)_
+- **Environment variable**: `DD_TRACE_SAMPLING_RULES` _(JSON)_
+- **Default value**: `[]`
+
+### Trace Flushing Period
+How often a batch of finished traces is sent to the Datadog Agent.
+
+- **TracerOptions member**: `int64_t write_period_ms` _(milliseconds)_
+- **Default value**: `1000` _(milliseconds)_
+
+### Operation Name
+The default operation name to associate with spans produced by the tracer.
+
+A span's operation name (sometimes just called "name" or "operation") indicates
+which of a service's functions the span represents.
+
+Operation name is often fixed for a given service, e.g. the "nginx" service
+entry spans might always have operation name "handle.request".
+
+Operation name is not to be confused with "resource," also known as endpoint,
+which is a trace tag.  Resource (endpoint) contains information about the
```

Comment:
`which is a trace tag` is not quite true, but OK.

Reply:
I see what you mean:

- https://github.com/DataDog/dd-opentracing-cpp/blob/80979a871759346bffa50f6fa215e3f87aed96b2/src/tags.cpp#L12
- https://github.com/DataDog/dd-opentracing-cpp/blob/43fd95339e6d8ef526238b5959d21877910b4db8/src/span.cpp#L145-L149

It's a non-tag span property that is set using the tagging API. I'll think of better wording.